bitnami/redis/templates/master/pdb.yaml

```diff
@@ -2,8 +2,8 @@
 Copyright Broadcom, Inc. All Rights Reserved.
 SPDX-License-Identifier: APACHE-2.0
 */}}
-
-{{- if and .Values.master.pdb.create (gt (int64 .Values.master.count) 0) (or (eq .Values.architecture "replication") (not .Values.sentinel.enabled)) }}
+{{- $pdb := coalesce .Values.pdb .Values.master.pdb }}
```

Comment:
Shouldn't it be the other way around? I mean, in this way `.Values.pdb` takes precedence over `.Values.master.pdb`?

Reply:
Good question. With current change If we have a values file coming from an older version with `pdb` defined, this will be used. If we call the coalesce function in the opposite way, the default values will be always applied and the `pdb` value will be ignored

Reply:
Newer values should take precedence over older ones. That's what we always do with deprecated values.

It's true that the values we've deprecated before were usually unset by default. That said, it's the user responsibility to update their values according the parameters available in newer versions and it would be very unusual to give priority to parameter that no longer exists.

Reply:
I agree with your idea in general but in this particular case your approach won't work, because `master.pdb` is set to an object by default and this will take precedence over `pdb` unless the user explicitly remove that object in the values.

Reply:
ahh! Good point, it makes sense